Fix for runtime 2022.1.0.482+

babaganosch opened the pull request:
Some minor changes to allow building with runtime 2022.1.0.482

* Temporary folder now uses login instead of username
* Output file is now named game.ios and game.yydebug (at least on osx?)

YellowAfterlife committed:
Fixed builder using _ around the appdata folder name, organized GMAC args better.

YellowAfterlife committed:
Slightly safer userpath/temppath resolution.

YellowAfterlife committed:
Added ability to set preferred runtime version on per-project basis.

YellowAfterlife committed:
Make use of config files instead of localStorage and a slightly better algorithm for picking temp drive letters

YellowAfterlife committed:
Added "Fork arguments override" to project properties

YellowAfterlife committed:
Hold up, did builder always pass fork arguments even for non-fork?

YellowAfterlife committed:
Split off preferences/project properties into their own files with ES6 classes

YellowAfterlife committed:
Similarly, relocate compiler invocation into a separate file

YellowAfterlife committed:
Similarly, relocate Windows virtual drive (subst management)

YellowAfterlife committed:
Split off output tab and added an option for displaying fork output in a separate tab.

YellowAfterlife committed:
Fixed current working directory mismatching with what GMS2 IDE would set when running.

YellowAfterlife committed:
Clear the aside-output when opening the first fork

YellowAfterlife committed:
Copy steam_api.dll to temp directory if steam_appid.txt is in the included files

YellowAfterlife committed:
Fixed project properties reading wrong

YellowAfterlife committed:
Added an option to show "Run & Fork" in main menu; keyboard shortcuts are now global

YellowAfterlife committed:
Catch stderr output from compiler (which contains your syntax errors)

YellowAfterlife committed:
Added a project option to override Steam App ID

YellowAfterlife committed:
Added icons to the menu items

YellowAfterlife committed:
Correctly change title of `chrome-tab-title-text`

YellowAfterlife committed:
Support extra.gml and replace.gml when loading per-platform API entries

YellowAfterlife committed:
Fixed some operations "leaking" temporary drives

YellowAfterlife committed:
Fixed an error when Runner exits with *no* status code

YellowAfterlife committed:
Added an option to [not] use virtual drives on Windows

YellowAfterlife committed:
Correction: fix up temp path if it doesn't end on a \/

babaganosch committed:
Change name of temporart folder from username to login name, change name of output files to 'game'

YellowAfterlife committed:
Fixed incompatibility with 2022.x runtime due to um.json format changes

brady-goldsworthy commented:
```
Temporary folder now uses login instead of username
```
This first change fixed builder on my local windows machine using the newer runtimes.


```
Output file is now named game.ios and game.yydebug (at least on osx?)
```
I Made this change on my local windows machine and builder stopped working, so it must be osx specific

Thanks @babaganosch!!

nommiin committed:
clarified usage steps

since this fork added new files, the mention of extracting specific files is sorta weird and confusing... so this just clears that up to avoid confusion

YellowAfterlife committed:
Merge pull request #1 from nommiin/patch-1

Clarified usage steps in README

YellowAfterlife committed:
Show Preferences inline

YellowAfterlife committed:
Fixed compatibility with GameMaker 2022.500.x betas

YellowAfterlife committed:
Merge remote-tracking branch 'origin/master'

babaganosch committed:
Merge remote-tracking branch 'upstream/master'

babaganosch committed:
Fix for osx runtime 2022.1.0.482+

Minor fixes to enable running the builder on osx. This commit does not however enable building on runtimes built with .NET6, more work is needed for that.

babaganosch committed:
Remove warning from README
